eng/Get-ServiceModules.ps1

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env pwsh
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+    .Synopsis
+    Gets the modules that are built from the given service directory and SDK type.
+
+    .Parameter ServiceDirectory
+    Name of the service being built. For example, "eventhubs".
+    .Parameter SdkType
+    Type of SDK to find modules for.
+
+    .Example
+    .\eng\Get-ServiceModules "eventhubs" "client"
+    Will output "azure-messaging-eventhubs,azure-messaging-eventhubs-checkpointstore-blob" because
+    those are modules that are built by the 'client' SDK type.
+
+    .Example
+    .\eng\Get-ServiceModules "eventhubs" "data"
+    Will output "microsoft-azure-eventhubs,microsoft-azure-eventhubs-eph,microsoft-azure-eventhubs-extensions"
+    because those are modules that are built by the 'data' SDK type.
+#>
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidateNotNullOrEmpty()]
+    [string]$ServiceDirectory,
+
+    [Parameter(Mandatory = $true, Position = 1)]
+    [ValidateSet("data", "client")]
+    [string]$SdkType
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey("ErrorAction")) {
+    $ErrorActionPreference = "Stop"
+}
+
+switch($SdkType) {
+    "client" {
+        $parentArtifactId = "azure-client-sdk-parent"
+        break
+    }
+    "data" {
+        $parentArtifactId = "azure-data-sdk-parent"
+        break
+    }
+    default {
+        throw "'$SdkType' is not supported."
+    }
+}
+
+# If it is not a module that is built by pom.client.xml, we should skip it.
+$clientModules = New-Object System.Collections.Generic.HashSet[string]
+[xml]$clientPom = Get-Content -Path (Join-Path "$PSScriptRoot/../" "pom.client.xml")
```

Comment:
Hard coded relative path.

Reply:
I saw this in other build scripts. Suggestions?

Reply:
If we must take a dependency on ```pom.client.xml``` then I'd prefer to pass the path in since I know its location is going to change. Preferably I would hope that we don't need to worry what the parent POM is. I mean in effect don't we just want to build all the modules that are referenced by ```pom.service.xml``` that sit under ```sdk/[servicedir]```? Why do we need to consult the parent at all?

Reply:
The reason for this is this. We were getting the error that it didn't know about module `azure-core-management`.
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/pom.service.xml#L11-L13